Routing/Loader/Reader/RestActionReader.php

```diff
@@ -203,6 +203,14 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
 
                 if (!empty($annoMethods)) {
                     $methods = $annoMethods;
+
+                    // workaround for new method annotations that are not supported in method
+                    // names until FOSRestBundle 2.0 for BC reasons (if this check wasn't added
+                    // the method name would have been appended to the generated path)
+                    if (!in_array(strtolower($annotation->getMethod()), $this->availableHTTPMethods)) {
+                        array_pop($urlParts);
+                        $path = implode('/', $urlParts);
+                    }
```

Comment:
This is the only change that we need to make here to keep BC (the HTTP method is only allowed to be different when the route is created due to an annotation).